packages/styles/src/timepicker.scss

```diff
@@ -0,0 +1,188 @@
+    height: $fd-time-picker-clock-face;
+    text-align: center;
+
+    &--compact {
+      width: $fd-time-picker-clock-face-compact;
+      height: $fd-time-picker-clock-face-compact;
+
+      .#{$block}__number {
+        width: $fd-time-picker-element-size-compact;
+        height: $fd-time-picker-element-size-compact;
+        line-height: $fd-time-picker-element-size-compact;
+      }
+
+      .#{$block}__item {
+        width: $fd-time-picker-element-size-compact;
+      }
+    }
+  }
+
+  &__clock {
+    @include fd-reset();
+
+    padding: 0.5625rem;
+    position: relative;
+
+    &--inner {
+      .#{$block}__label::before {
+        background: var(--sapLegend_WorkingBackground);
+      }
+    }
+  }
+
+  &__label {
+    @include fd-reset();
+
+    width: auto;
+    display: block;
+    position: relative;
+    text-align: center;
+    border-radius: 100%;
+    background: var(--sapLegend_WorkingBackground);
+
+    &::before {
+      @include fd-reset();
+      @include fd-flex-center();
+      @include fd-set-equal-positions(2.75rem);
+
+      position: absolute;
+      border-radius: 100%;
+      content: attr(data-label);
+      color: var(--sapContent_LabelColor);
+    }
+
+    &::after {
+      content: '';
+      display: block;
+      padding-bottom: 100%;
+    }
+  }
+
+  &__item {
+    @include fd-reset();
+
+    position: absolute;
+    top: 0;
+    left: 50%;
+    width: $fd-time-picker-element-size;
+    height: 100%;
+    user-select: none;
+    display: inline-block;
+  }
+
+  &__tick {
+    @include fd-reset();
+
+    display: block;
+    border-radius: 100%;
+    margin: 0 auto 0.25rem auto;
+    background: var(--sapField_BorderColor);
+
+    &--hour {
+      width: 0.25rem;
+      height: 0.3125rem;
+    }
+
+    &--minute {
+      width: 0.125rem;
+      height: 0.1875rem;
+      margin-bottom: 0.375rem;
+    }
+
+    &--selected {
+      margin: 0 auto;
+      width: 0.25rem;
+      height: 0.5625rem;
+      background: var(--sapButton_Selected_Background);
+      border: 0.0625rem solid var(--sapButton_Selected_BorderColor);
+    }
+  }
+
+  &__number {
+    @include fd-reset();
+    @include fd-flex-center();
+
+    width: $fd-time-picker-element-size;
+    height: $fd-time-picker-element-size;
+    line-height: $fd-time-picker-element-size;
+    text-align: center;
+    border-radius: 100%;
+    vertical-align: top;
+    border: 0.0625rem solid transparent;
+
+    &--hover {
+      background: var(--sapList_Hover_Background);
+    }
+
+    &--invisible {
+      visibility: hidden;
+    }
+
+    &--selected {
+      color: var(--sapButton_Selected_TextColor);
+      background: var(--sapButton_Selected_Background);
+      border-color: var(--sapButton_Selected_BorderColor);
+    }
+  }
+}
+
+$i: 6;
+
+@while $i < 361 {
+  .#{$block}__item--deg#{$i} {
+    transform: translate(-50%) rotate(#{$i}deg);
+
+    .#{$block}__number {
+      transform: rotate(-#{$i}deg);
+    }
+  }
+  $i: $i + 6;
+}
```

Comment:
CSS variables would take a lot less space for this one

Reply:
But I would go with [Data attributes](https://css-tricks.com/a-complete-guide-to-data-attributes/#aa-using-data-attributes-visually)

Reply:
could you be more specific?